clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3660,7 +3686,7 @@ void handleResponse(AbstractResponse abstractResponse) {
                     // We set member.id to empty here explicitly, so that the lookup will succeed as user doesn't
                     // know the exact member.id.
                     memberErrors.put(new MemberIdentity()
-                                         .setMemberId(JoinGroupRequest.UNKNOWN_MEMBER_ID)
+                                         .setMemberId(memberResponse.memberId())
```

Comment:
I could see this doesn't hold true for a plain static member removal. Let's discuss why skipping the individual member check in `RemoveMembersFromConsumerGroupResult` makes sense over there.

Reply:
For removing static members, this still true because we put memberId as `""` in the request, and the server will also response with the same request field. (Verified `GroupCoordinator#handleLeaveGroup`)
For removing dynamic members, we need this change to know the memberId for the caller.
I suppose the `individual check` here is just to check the response against the members to be removed(for `removeAll` scenario)? Previously I thought of putting all members got from `KafkaAdminClient#getMembersFromGroup` in the RemoveMembersFromConsumerGroupResult for checking, but in `removeAll` scenario, we get members as  `MemberIdentity` which cannot be converted back to `MemberToRemove`, so I'm hesitate to do in this way

Reply:
Not sure if I understand the change. Also not sure if I can follow the comments. Can you elaborate?